readme.md

```diff
@@ -38,10 +38,17 @@ The source object to copy properties from.
 
 #### mapper
 
-Type: `(sourceKey, sourceValue, source) => [targetKey, targetValue, mapperOptions?] | mapObjectSkip`
+Type: `(sourceKey, sourceValue, source, path) => [targetKey, targetValue, mapperOptions?] | mapObjectSkip`
 
 A mapping function.
 
+##### path
+
+Type: `string[]|undefined`\
+
+If `deep === true`, this is the sequence of keys to reach the current value from the `source`;
```

Comment:
I think it's simpler to make it an empty array when `deep: false`.

Reply:
But then we would be creating a new unnecessary object for every iterarion.  The `path` parameter will probably not even be read by the user because the user themselves decide whether the mapping will be deep or not when they call 'mapObject()`. Are you sure?

Reply:
> But then we would be creating a new unnecessary object for every iterarion. 

You are assuming the JS engine is not able to optimize that out.

Reply:
The alternative is to complicate the TypeScript types. Your current types are not correct. You would have to conditionalize the `path` type based on the value of the `deep` option.

Reply:
Done.